tools/bazar/services/EntryManager.php

```diff
@@ -454,6 +459,72 @@ public function update($tag, $data, $semantic = false, $replace = false)
         return $data;
     }
 
+    /**
+     * Replace the field values which are restricted at reading and writing. These values must be loaded to save them
+     * without user modification.
+     * As the fields are rectricted at reading, the right must be bypassed to load them.
+     *
+     * @param array $data the provided data to update
+     * @return array the data with the restricted values added
+     * @throws Exception
+     */
+    protected function assignRestrictedFields(array $data)
+    {
+        // not possible to init the formManager in the constructor because of circular reference problem
+        $form = $this->wiki->services->get(FormManager::class)->getOne($data['id_typeannonce']);
+
+        // check if there are some restricted fields at both reading and writing
+        $restrictedFields = [];
+        foreach ($form['prepared'] as $field) {
+            if ($field instanceof BazarField) {
+                $propName = $field->getPropertyName();
+                if (!$field->canRead($data) && !$field->canEdit($data)) {
+                    $restrictedFields[] = $propName;
+                }
+                // TODO when entry update from the form will be done by partiel update ($replace = true), copy the field value when canRead & !canEdit
+            }
+        }
+
+        if (!empty($restrictedFields)) {
+            // if there are some restricted fields, load the previous data by bypassing the rights
+            $previousData = $this->getOne($data['id_fiche'], false, null, false, true);
+
+            // get the value of the restricted fields in the previous data
+            foreach ($restrictedFields as $propName) {
+                $data[$propName] = $previousData[$propName] ?? null;
```

Comment:
```suggestion
                if (isset($previousData[$propName])){
                    $data[$propName] = $previousData[$propName] ;
                } elseif (isset($data[$propName])) {
                     unset($data[$propName]);
                }
```

Il est important selon moi de ne pas créer des clés vides

Reply:
je suis d'accord, je crois d'ailleurs que je l'avais fait dans la branche partialUpdate.
bonne idée aussi de rajouter le cas dont on parlais tout à l'heure. il faudrait juste qu'on rajoute un commentaire pour l'expliciter

Reply:
Le code ne se suffit pas à lui-même ?

Reply:
ça ne coûte rien de le spécifier car nul part on met qu'on fait en sorte de ne pas mettre de null associé à une clé.

Reply:
mince ce n'était pas clair. Le commentaire dont je parlais à la base était en fait pour la deuxième partie. 
Car ce deuxième test est uniquement pour le cas où il y a une variable injectée dans le $_POST et que la donnée précédente n'a pas la clé, et sans commentaire, je ne vois pas comment quelqu'un peut le deviner sans commentaire.  
Je vais le rajouter et tu me diras si c'est ok.

Reply:
Je préfère que tu me fasses une proposition de commentaire en mode commentaire d'une revue plutôt qu'en faisant un commit. Possible pour toi ?

Reply:
oups j'ai vu seulement là ton commentaire.